### PR TITLE
[hardknott] merge reliable meta-nilrt layer audit commits

### PR DESCRIPTION
**Pull request by amstewart**
This PR merges commits from the meta-nilrt layer audit which I judge to be "reliable", in the sense that they are unlikely to break any image functionality. Specifically, it...
1. moves `ntp` from `pkggrp-ni-extra` to `pkggrp-ni-desirable`
2. removes the opencv bbappend - since its only purpose was to fix recipe compilation, which has since been fixed upstream.
3. enhances the `salt` recipe subpackage summaries and descriptions
4. merge the `trousers` recipe files into one recipe (trivial copy-paste job)
5. upgrades the `base-files-nilrt` package from `1.0` (default) to `2.0`
6. removed ARM and nxg logic from some recipes
7. adds some missing RDEPENDS to recipes.

NI AZDO: https://dev.azure.com/ni/DevCentral/_workitems/edit/2116516

# Testing
* Built the core feed locally and verified
    * that the `salt` subpackage descriptions are updated to their new values.
    * `base-files-nilrt` installs the same file paths, after the ARM logic removal.
    * `initscripts-nilrt` installs exactly the same content after the logic removal/simplification.

# Maintainership
This PR is based on #461.

### CODE REVIEW

File: recipes-support/salt/salt_3000.2.bb [hunk 1]
```diff
@@ -122,8 +125,8 @@ INITSCRIPT_PARAMS:${PN}-api = "defaults"
 FILES:${PN}-bash-completion = "${sysconfdir}/bash_completion.d/${PN}-common"
 
 # salt-cloud
-SUMMARY:${PN}-cloud = "public cloud VM management system"
-DESCRIPTION:${PN}-cloud = "provision virtual machines on various public clouds via a cleanly controlled profile and mapping system."
+SUMMARY:${PN}-cloud = "${SUMMARY}; public cloud VM management system"
```
Comment (shruthi-ravi) [inline recipes-support/salt/salt_3000.2.bb:128]: It is a little odd that all of these summaries now start with `Salt - remote execution manager`. It's a little misleading? Could you instead say add a "for salt" suffix to the summaries?

File: recipes-support/salt/salt_3000.2.bb [hunk 2]
```diff
@@ -136,8 +139,8 @@ FILES:${PN}-cloud = "\
 "
 
 # salt-common
-SUMMARY:${PN}-common = "shared libraries that salt requires for all packages"
-DESCRIPTION:${PN}-common ="${DESCRIPTION_COMMON} This particular package provides shared libraries that \
+SUMMARY:${PN}-common = "${SUMMARY}; shared libraries that salt requires for all packages"
+DESCRIPTION:${PN}-common ="${DESCRIPTION} This particular package provides shared libraries that \
```
Comment (shruthi-ravi) [inline recipes-support/salt/salt_3000.2.bb:143]: Same for the description. Too much redundant and verbose information for every subpackage.

File: recipes-core/busybox/busybox_1.33.%.bbappend [hunk 3]
```diff
@@ -1,50 +1,51 @@
...
+			file://ifplugd.action \
+			file://ifplugd.conf \
+			file://zcip.script \
+			file://busybox-acpid \
+			file://acpid.conf \
+			file://acpid_poweroff.sh \
+			file://acpid-logrotate.conf \
+			file://zcip-allow-action-script-to-reject-chosen-IP.patch \
+			file://login-utilities.cfg \
+			file://udhcpd.wlan0.conf"
```
Comment (mike-petersen-ni) [inline recipes-core/busybox/busybox_1.33.%.bbappend:14]: Multi-tabs

```suggestion
	file://busybox-ifplugd \
	file://ifplugd.action \
	file://ifplugd.conf \
	file://zcip.script \
	file://busybox-acpid \
	file://acpid.conf \
	file://acpid_poweroff.sh \
	file://acpid-logrotate.conf \
	file://zcip-allow-action-script-to-reject-chosen-IP.patch \
	file://login-utilities.cfg \
	file://udhcpd.wlan0.conf"
```

File: recipes-core/images/includes/nilrt-proprietary.inc [hunk 4]
```diff
@@ -1,48 +1,48 @@
 # NI infrastructure packages to be installed in the image
 
 NI_PROPRIETARY_SAFEMODE_PACKAGES = "\
-        ni-software-installation-daemon \
+		ni-software-installation-daemon \
```
Comment (mike-petersen-ni) [inline recipes-core/images/includes/nilrt-proprietary.inc:4]: Multi-tabs

```suggestion
	ni-software-installation-daemon \
```

File: recipes-core/images/includes/nilrt-proprietary.inc [hunk 5]
```diff
@@ -1,48 +1,48 @@
...
+		ni-webservices-webserver-support \
+		ni-wif-landingpage \
+		ni-wifibledd \
+		ni-wireless-ath6kl \
+		ni-wireless-cert-management-webservice \
+		ni-wireless-cert-storage \
+		nicurl \
+		nirtcfg \
+		nirtmdnsd \
+		nissl \
```
Comment (mike-petersen-ni) [inline recipes-core/images/includes/nilrt-proprietary.inc:39]: Multi-tabs

```suggestion
	libnitargetcfg \
	ni-auth \
	ni-auth-webservice \
	ni-avahi-client \
	ni-ca-certs \
	ni-rt-exec-webservice \
	ni-sdmon \
	ni-service-locator \
	ni-skyline-file-client \
	ni-skyline-message-client \
	ni-skyline-tag-client \
	ni-software-action-services \
	ni-software-installation-websvc \
	ni-ssl-webserver-support \
	ni-sysapi \
	ni-sysapi-remote \
	ni-sysapi-webservice \
	ni-sysmgmt-auth-utils \
	ni-system-webserver \
	ni-traceengine \
	ni-webdav-system-webserver-support \
	ni-webserver-libs \
	ni-webservices-webserver-support \
	ni-wif-landingpage \
	ni-wifibledd \
	ni-wireless-ath6kl \
	ni-wireless-cert-management-webservice \
	ni-wireless-cert-storage \
	nicurl \
	nirtcfg \
	nirtmdnsd \
	nissl \
```

File: recipes-core/images/includes/nilrt-proprietary.inc [hunk 6]
```diff
@@ -1,48 +1,48 @@
...
-        ni-arch-gen \
-        ni-sysmgmt-salt-minion-support \
-        ni-sysmgmt-sysapi-expert \
-        python3-ni-asset-discovery \
-        python3-ni-systemlink-sdk \
+		ni-arch-gen \
+		ni-sysmgmt-salt-minion-support \
+		ni-sysmgmt-sysapi-expert \
+		python3-ni-asset-discovery \
+		python3-ni-systemlink-sdk \
```
Comment (mike-petersen-ni) [inline recipes-core/images/includes/nilrt-proprietary.inc:47]: Multi-tabs

```suggestion
	ni-arch-gen \
	ni-sysmgmt-salt-minion-support \
	ni-sysmgmt-sysapi-expert \
	python3-ni-asset-discovery \
	python3-ni-systemlink-sdk \
```

File: recipes-core/util-linux/util-linux_2.%.bbappend [hunk 7]
```diff
@@ -1,18 +1,18 @@
...
 }
 
-pkg_postinst_ontarget_${PN}-hwclock () {
-        if [ ! -f /etc/natinst/safemode ]; then
-            setcap CAP_SYS_TIME+ep ${base_sbindir}/hwclock.util-linux
-        fi
+pkg_postinst:ontarget:${PN}-hwclock () {
+		if [ ! -f /etc/natinst/safemode ]; then
+			setcap CAP_SYS_TIME+ep ${base_sbindir}/hwclock.util-linux
+		fi
```
Comment (mike-petersen-ni) [inline recipes-core/util-linux/util-linux_2.%.bbappend:17]: Multi-tabs

```suggestion
	if [ ! -f /etc/natinst/safemode ]; then
		setcap CAP_SYS_TIME+ep ${base_sbindir}/hwclock.util-linux
	fi
```

File: recipes-ni/ni-systemimage/ni-systemimage.bb [hunk 8]
```diff
@@ -1,17 +1,22 @@
-SUMMARY = "A system imaging utility for NI LinuxRT safemodes"
+SUMMARY = "A system imaging utility for NI NILRT safemodes"
```
Comment (mike-petersen-ni) [inline recipes-ni/ni-systemimage/ni-systemimage.bb:1]: Same fixup as 03352049f248463435cdd22e0166e55b0f9b4f4a.

```suggestion
SUMMARY = "A system imaging utility for NILRT safemodes"
```

**Review by mike-petersen-ni — APPROVED** [review]

**Comment by amstewart** [comment]
These audit commits are going to be reorganized and resubmitted in a way which is easier to review.

**Pull request closed without merging**